packages/babel-generator/src/node/parentheses.ts

```diff
@@ -281,7 +292,34 @@ export function LogicalExpression(node: any, parent: any): boolean {
   }
 }
 
-export function Identifier(node: t.Identifier, parent: t.Node): boolean {
+export function Identifier(
+  node: t.Identifier,
+  parent: t.Node,
+  printStack: Array<t.Node>,
+): boolean {
+  // Non-strict code allows the identifier `let`, but it cannot occur as-is in
+  // certain contexts to avoid ambiguity with contextual keyword `let`.
+  if (node.name === "let") {
+    // Some contexts only forbid `let [`, so check if the next token would
+    // be the left bracket of a computed member expression.
+    const isFollowedByBracket =
+      t.isMemberExpression(parent, {
+        object: node,
+        computed: true,
+      }) ||
+      t.isOptionalMemberExpression(parent, {
```

Comment:
If the `parent` is an optional member expression with `optional: false`, then `let[` should follow an optional chain `foo?.`, in that case I don't think we should take care of `let`.

Reply:
It was mentioned at https://github.com/babel/babel/pull/13269#issuecomment-833532143. I think it's better to be safe now since we don't prevent `OptionalMemberExpression { optional: false, object: Identifier { "let" } }`